static/js/src/app/campaign_results.js

```diff
@@ -925,7 +925,26 @@ function refresh() {
     setRefresh = setTimeout(refresh, 60000)
 };
 
-
+function report_mail(rid, cid) {
+    var r = confirm("Mark this mail as reported?");
+    if (r == true) {
+        api.campaignId.get(cid).success((function(c) {
+            report_url = c.url;
+            report_url += report_url.endsWith("/") ? "" : "/";
+            report_url += "report?rid=" + rid;
+            $.ajax({
+                url: report_url,
+                method: "GET",
+                success: function(data) {
+                    return; // wait for response

```

Comment:
What do you think about moving the call to refresh here, that way it fires as soon as the `success` callback is fired?

Reply:
ofc, that will be a better code style! The timout / sleep function is important for the refresh to work, not the best solution i think, but works good. I will make a commit